src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -118,14 +110,55 @@ public async Task StartContainerAsync(IExecutionContext executionContext, object
                 }
             }
 
-#if !OS_WINDOWS
             executionContext.Debug($"Delete stale container networks from previous jobs");
             int networkPruneExitCode = await _dockerManger.DockerNetworkPrune(executionContext);
             if (networkPruneExitCode != 0)
             {
                 executionContext.Warning($"Delete stale container networks failed, docker network prune fail with exit code {networkPruneExitCode}");
             }
-#endif
+
+            // Create local docker network for this job to avoid port conflict when multiple agents run on same machine.
+            // All containers within a job join the same network
+            var containerNetwork = $"vsts_network_{Guid.NewGuid().ToString("N")}";
```

Comment:
set `var containerNetwork = $"vsts_network_{Guid.NewGuid().ToString("N")}";` as a private variable in this class.

Reply:
Nevermind. I see what you mean.